### ISSUE TITLE
multiple columns for postgres `on conflict` clause

### ISSUE THREAD

**Issue by ghost** [doug-martin/goqu#339]
**Is your feature request related to a problem? Please describe.**
I have a table with multiple columns being part of the primary key, and I get/insert default values 
```sql
insert into table_name (key1, key2) values ($1, $2) on conflict (key1, key2) do update set key1 = $1;
```

right now this is not possible to accomplish with goqu, since it only accepts a single column for `on conflict`.

**Describe the solution you'd like**
I would like a way to put multiple columns into `on conflict` clause.

**Describe alternatives you've considered**
it would be possible to do this by adding additional SQL query text manually.

**Dialect**
- [x] postgres

**Comment by abdusco** [comment]
Try using `goqu.Record`:

```golang
goqu.Insert("table").OnConflict(goqu.DoUpdate(
		"id",
		goqu.Record{
			"id": goqu.I("new.id"),
			"name": goqu.I("new.name"),
		},
	))
```

which produces

```sql
... ON CONFLICT (id) 
DO UPDATE SET "id"="new"."id","name"="new"."name"
```

**Comment by chapmanc** [comment]
Also having this issue and I am not sure if @abdusco's suggestion addresses the issue. I want to check a conflict on three columns. Open to any suggestions here but being forced to go literal sql I think as well. 
```
... ON CONFLICT (key1, key2, key3) 
DO UPDATE SET ... 
```

**Comment by abdusco** [comment]
@chapmanc If you already have an index, you should be able to use `... ON CONFLICT ON CONSTRAINT ...` syntax

https://www.postgresql.org/docs/current/sql-insert.html#SQL-ON-CONFLICT

```golang
q := goqu.Insert("table").OnConflict(goqu.DoUpdate(
		"on constraint my_index",
		goqu.Record{
			"id":   goqu.I("new.id"),
			"name": goqu.I("new.name"),
		},
	))
```

however, goqu does not support multiple columns otherwise.

**Comment by chapmanc** [comment]
Yeah unfortunately it's a poorly constructed table. I did discover I could put a comma separated set of columns and get the equivalent sql although it doesn't feel all that great: 

```go
goqu.DoUpdate("key1, key2, key3", goqu.Record{ "updated_at": time.Now()})
```

**Comment by spiralw** [comment]
looks like the `on constraint my_index` will work for me, thanks. i'll keep this issue open for the actual feature request

**Comment by abdusco** [comment]
@spiralw I wouldn't hold my breath. The library looks abandoned. I even emailed the author for getting contribution rights a couple months ago, but haven't received any response.
